crates/ln-dlc-node/src/tests/just_in_time_channel/create.rs

```diff
@@ -158,9 +190,16 @@ pub(crate) async fn send_interceptable_payment(
         payee.connect(coordinator.info).await?;
     }
 
-    payee
-        .wait_for_payment_claimed(invoice.payment_hash())
-        .await?;
+    if let Err(e) = payee.wait_for_payment_claimed(invoice.payment_hash()).await {
```

Comment:
:thinking: Can we please not add any more branches to these tests? If we want to reuse code and have a different assertion we should just compose things differently.

These enums that activate different test branches are a nightmare for test clarity and they're not strictly necessary.

I know you didn't add the enum, but I would aim to get rid of it rather than continue to extend it.

Reply:
I agree with your general sentiment, and find TestPaths really annoying.

However, in this case the `TestPath` is used exactly in the same function, so I don't think it makes it any worse.

I will create a ticket to address this technical debt, as refactoring is as such out of scope for this PR.

Reply:
Fair enough! We obviously don't have to do this right now, but hopefully soon :pray: